plugin/src/main/java/org/opensearch/ml/settings/MLCommonsSettings.java

```diff
@@ -117,6 +117,27 @@ private MLCommonsSettings() {}
     public static final Setting<Boolean> ML_COMMONS_REMOTE_INFERENCE_ENABLED = Setting
         .boolSetting("plugins.ml_commons.remote_inference.enabled", true, Setting.Property.NodeScope, Setting.Property.Dynamic);
 
+    public static final Setting<Boolean> ML_COMMONS_REMOTE_INFERENCE_RETRY_ENABLED = Setting
+        .boolSetting("plugins.ml_commons.remote_inference.retry_enabled", true, Setting.Property.NodeScope, Setting.Property.Dynamic);
+
+    public static final Setting<Integer> ML_COMMONS_REMOTE_INFERENCE_RETRY_BACKOFF_MILLIS = Setting
+        .intSetting(
+            "plugins.ml_commons.remote_inference.retry_backoff_millis",
+            100,
+            1,
+            Setting.Property.NodeScope,
+            Setting.Property.Dynamic
+        );
+
+    public static final Setting<Integer> ML_COMMONS_REMOTE_INFERENCE_RETRY_TIMEOUT_SECONDS = Setting
+        .intSetting(
+            "plugins.ml_commons.remote_inference.retry_timeout_seconds",
+            30,
```

Comment:
Was there any end-to-end tests using the _Bulk API to verify the impact to performance using the 30s retry? For example, having a sageMaker endpoint with t2.medium instance type and invoking it with 200 docs with 3 nodes, etc.

Reply:
I tried with g4dn.xlarge + 3 nodes and multiple bulk settings, including 12 clients with size 10 and 1 clients with size 2000. ([ref](https://github.com/opensearch-project/ml-commons/issues/2438#issuecomment-2117001088))
If the workload within the capacity of the endpoint, i.e. the endpoint can process all requests within the connectionAcquisitionTimeout, I didn't see any throttling exception in the benchmark

Reply:
And I think for `t2.medium` it works the same, the only difference is the endpoint throughput capacity

Reply:
Consider changing this to milliSeconds, the inference could be in reading path which is sensitive to latency.

Reply:
But change to millis is only about metric, it doesn't change the actual wait time, right?

Since default hyperparameters should work for general scenerios, I believe the top priority is to make the connector run stably under reasonable workload, don't throw throttling exceptions that makes data lost. That's why I set it a rather long timeout value of 30s.
For latency sensitive use case, they can configure the rest request level timeout to achieve the same effect

Reply:
We don't have a mechanism that to identify the request is from query or ingestion, and since we're targeting to solve ingestion issue, I think it's fine to leave it what it is now.